examples/scene/volume_depth.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+# vispy: gallery 10:200:5
+"""
+Render volumes with depth_test
+================================
+
+Controls:
+* p  - toggle camera fov (perspective) between 0 and 60
+"""
+import numpy as np
+
+from vispy import app, scene
+from vispy.color import get_colormap
+
+
+# Prepare canvas
+canvas = scene.SceneCanvas(keys='interactive', size=(800, 600), show=True)
+
+# Set up a viewbox to display the image with interactive pan/zoom
+view = canvas.central_widget.add_view()
+
+# Create the volume visual
+im_data = np.zeros((30, 30, 30))
+im_data[5:25, 5:25, 5:25] = 1
+blue_cmap = get_colormap('blues')
+volume_1 = scene.visuals.Volume(
+    im_data,
+    cmap=blue_cmap,
+    parent=view.scene,
+    threshold=0.5,
```

Comment:
This example is actually failing in the tests because of these warnings:

```
Example /scene/volume_depth.py failed (0):
----------------------------------------------------------------------
/home/runner/work/vispy/vispy/vispy/gloo/texture.py:28: UserWarning: GPUs can't support floating point data with more than 32-bits, precision will be lost due to downcasting to 32-bit float.
  warnings.warn(F64_PRECISION_WARNING)
WARNING: Program has unset variables: {'u_threshold'}
```

The 64-bit float one should be easy to fix if you set `dtype=np.float32` in your `np.zeros` call. The other one about threshold, I'm not sure why threshold would be unset if the `iso` method uses it (does it?).

Reply:
Thanks for the tip. I've changed the data dtype to `np.float32` and that fixed the precision warning.

I'm a bit confused by the `u_threshold` warning. My understanding is that `u_threshold` is the threshold for finding the surface in ` iso` rendering mode. I tested the example on `main` and I get the same warning, so I think it's a pre-existing issue.

Reply:
If we end up removing the example, I can make an issue for the `u_threshold` warning.

Reply:
I think this is the result of some weird ordering. In the `Volume.__init__` method, the `self.method` is set first, then the `self.threshold` is set. And `self.update()` is called at the end of each method. Because you set the method to `iso` immediately, it doesn't have a threshold set and tries to draw immediately (I guess) and produces the warning. Looking at the code, even if you used the default `method` first then set it to `iso` later, I think you'd still get the warning no matter what you did because of the way things are set. It looks like it could definitely be improved, but don't worry about it in this PR if others think the example should get removed. If the example does stay then we'll have to think of something.